libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -32,6 +32,60 @@ using namespace solidity;
 using namespace solidity::langutil;
 using namespace solidity::frontend;
 
+namespace
+{
+
+
+struct TypeComp
+{
+	bool operator()(Type const* lhs, Type const* rhs) const
+	{
+		solAssert(lhs && rhs, "");
+		return lhs->richIdentifier() < rhs->richIdentifier();
+	}
+};
+using TypeSet = std::set<Type const*, TypeComp>;
+
+class OversizedTypeChecker
+{
+public:
+	OversizedTypeChecker(Type const& _type)
+	{
+		visit(_type, true);
+	}
+
+	vector<Type const*> oversizedTypes()
+	{
+		return util::convertContainer<vector<Type const*>>(std::move(m_oversizedTypes));
+	}
+
+private:
+	void visit(Type const& _type, bool _needsSizeCheck)
+	{
+		if (_needsSizeCheck && _type.storageSizeUpperBound() >= bigint(1) << 64)
```

Comment:
Why 2^64?

The type system (https://github.com/ethereum/solidity/blob/5f8299b379a1f4d99f491a0b91fbbda853e580e5/libsolidity/ast/Types.cpp#L1742) seems to be used unsigned (2^32).

If you change this to 1<<32, I think this PR will also fix #8382, if so, please include the test case https://github.com/ethereum/solidity/issues/8382#issuecomment-642417741

Reply:
It is a different error. This PR is about storage size, the "array size does not fit unsigned" is for arrays for calldata and memory.